sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -192,39 +196,214 @@ private void sendBody(HttpRequest httpRequest, Object progressReporter, HttpURLC
+            ServerSentEventListener listener = httpRequest.getServerSentEventListener();
+            RetrySSEResult retrySSEResult;
+            if (connection.getErrorStream() == null) {
+                if (isTextEventStream(responseHeaders) && listener != null) {
+                    try {
+                        BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+                        retrySSEResult = processBuffer(reader, listener);
+                        if (retrySSEResult != null) {
+                            retryExceptionForSSE(retrySSEResult, listener, httpRequest);
+                        }
+                    } catch (IOException e) {
+                        throw LOGGER.logThrowableAsError(new RuntimeException(e));
+                    }
+                }
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders);
+            } else {
+                AccessibleByteArrayOutputStream outputStream = getAccessibleByteArrayOutputStream(connection);
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders,
+                    BinaryData.fromByteBuffer(outputStream.toByteBuffer()));
+            }
+        } catch (IOException e) {
+            throw LOGGER.logThrowableAsError(new RuntimeException(e));
+        } finally {
+            connection.disconnect();
+        }
+    }
+
+    private Headers getResponseHeaders(HttpURLConnection connection) {
+        Map<String, List<String>> hucHeaders = connection.getHeaderFields();
+        Headers responseHeaders = new Headers((int) (hucHeaders.size() / 0.75F));
+        for (Map.Entry<String, List<String>> entry : connection.getHeaderFields().entrySet()) {
+            if (entry.getKey() != null) {
+                responseHeaders.add(HeaderName.fromString(entry.getKey()), entry.getValue());
+            }
+        }
+        return responseHeaders;
+    }
+
+    private boolean isTextEventStream(Headers responseHeaders) {
+        return responseHeaders.get(HeaderName.CONTENT_TYPE).getValue().equals(ContentType.TEXT_EVENT_STREAM);
+    }
 
-            Map<String, List<String>> hucHeaders = connection.getHeaderFields();
-            Headers responseHeaders = new Headers((int) (hucHeaders.size() / 0.75F));
+    private static AccessibleByteArrayOutputStream getAccessibleByteArrayOutputStream(HttpURLConnection connection) throws IOException {
+        AccessibleByteArrayOutputStream outputStream = new AccessibleByteArrayOutputStream();
 
-            for (Map.Entry<String, List<String>> entry : connection.getHeaderFields().entrySet()) {
-                if (entry.getKey() != null) {
-                    responseHeaders.add(HeaderName.fromString(entry.getKey()), entry.getValue());
+        try (InputStream errorStream = connection.getErrorStream();
+             InputStream inputStream = (errorStream == null) ? connection.getInputStream() : errorStream) {
```

Comment:
This method is called only when the errorStream is not null in `receiveResponse` method. So, we don't need to check for null here again.